inc/providers/class-wp-sitemaps-posts.php

```diff
@@ -141,19 +155,67 @@ public function max_num_pages( $post_type = '' ) {
 			return 0;
 		}
 
-		$query = new WP_Query(
+		/**
+		 * Filters the max number of pages before it is generated.
+		 *
+		 * Passing a non-null value will effectively short-circuit the generation,
+		 * returning that value instead.
+		 *
+		 * @since 5.5.0
+		 *
+		 * @param int $max_num_pages The maximum number of pages. Default null.
+		 * @param string $post_type Post type name.
+		 */
+		$max_num_pages = apply_filters( 'wp_sitemaps_taxonomies_pre_max_num_pages', null, $post_type );
```

Comment:
this should be `wp_sitemaps_posts_pre_max_num_pages`

Reply:
That's what I get for copy/pasting code 😅